File: ontrack-service/src/main/java/net/nemerosa/ontrack/service/security/SecurityConfiguration.java
package net.nemerosa.ontrack.service.security;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.AuthenticationProvider;
import org.springframework.security.authentication.ProviderManager;

import java.util.Arrays;

@Configuration
public class SecurityConfiguration {

    @Autowired
    @Qualifier("password")
    private AuthenticationProvider passwordAuthenticationProvider;

    @Bean
    public AuthenticationManager authenticationManager() {
        return new ProviderManager(
                Arrays.asList(
                        passwordAuthenticationProvider
                )
        );
    }

}

File: ontrack-service/src/main/java/net/nemerosa/ontrack/service/security/SecurityConfiguration.kt
package net.nemerosa.ontrack.service.security

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.authentication.AuthenticationProvider
import org.springframework.security.authentication.ProviderManager


@Configuration
class SecurityConfiguration(
        private val providers: List<AuthenticationProvider>
) {

    @Bean
    fun authenticationManager() = ProviderManager(providers)

}